src/command_modules/azure-cli-backup/azure/cli/command_modules/backup/custom.py

```diff
@@ -0,0 +1,755 @@
+                                                                           custom_headers=_get_custom_headers())
+
+
+def _job_in_progress(job_status):
+    return job_status == JobStatus.in_progress.value or job_status == JobStatus.cancelling.value
+
+# List Utilities
+
+
+def _get_list_from_paged_response(obj_list):
+    return list(obj_list) if isinstance(obj_list, Paged) else obj_list
+
+
+def _get_one_or_many(obj_list):
+    return obj_list[0] if len(obj_list) == 1 else obj_list
+
+
+def _get_filter_string(filter_dict):
+    filter_list = []
+    for k, v in sorted(filter_dict.items()):
+        filter_segment = None
+        if isinstance(v, str):
+            filter_segment = "{} eq '{}'".format(k, v)
+        elif isinstance(v, datetime):
+            filter_segment = "{} eq '{}'".format(k, v.strftime('%Y-%m-%d %I:%M:%S %p'))  # yyyy-MM-dd hh:mm:ss tt
+        if filter_segment is not None:
+            filter_list.append(filter_segment)
+    filter_string = " and ".join(filter_list)
+    return None if not filter_string else filter_string
+
+
+def _get_query_dates(end_date, start_date):
+    query_start_date = None
+    query_end_date = None
+    if start_date and end_date:
+        query_start_date = start_date
+        query_end_date = end_date
+    elif not start_date and end_date:
+        query_end_date = end_date
+        query_start_date = query_end_date - timedelta(days=30)
+    elif start_date and not end_date:
+        query_start_date = start_date
+        query_end_date = query_start_date + timedelta(days=30)
+    return query_end_date, query_start_date
+
+# JSON Utilities
+
+
+def _get_container_from_json(client, container):
+    return _get_object_from_json(client, container, 'ProtectionContainerResource')
+
+
+def _get_vault_from_json(client, vault):
+    return _get_object_from_json(client, vault, 'Vault')
+
+
+def _get_vm_from_json(client, vm):
+    return _get_object_from_json(client, vm, 'VirtualMachine')
+
+
+def _get_policy_from_json(client, policy):
+    return _get_object_from_json(client, policy, 'ProtectionPolicyResource')
+
+
+def _get_item_from_json(client, item):
+    return _get_object_from_json(client, item, 'ProtectedItemResource')
+
+
+def _get_job_from_json(client, job):
+    return _get_object_from_json(client, job, 'JobResource')
+
+
+def _get_recovery_point_from_json(client, recovery_point):
+    return _get_object_from_json(client, recovery_point, 'RecoveryPointResource')
+
+
+def _get_or_read_json(json_or_file):
+    json_obj = None
+    if is_json(json_or_file):
+        json_obj = json.loads(json_or_file)
+    elif os.path.exists(json_or_file):
+        with open(json_or_file) as f:
+            json_obj = json.load(f)
+    if json_obj is None:
+        raise ValueError(
+            """
+            The variable passed should be in JSON format supplied by az backup CLI commands.
+            Make sure that you use relevant az backup show commandlets output and the output is json
```

Comment:
'commandlets' -> 'commands'.
'commandlets' is very 'PowerShelly'.

Reply:
Thanks for catching this; got missed!